creator-node/src/routes/files.js

```diff
@@ -318,6 +320,62 @@ const getDirCID = async (req, res) => {
   }
 }
 
+/**
+ * Perform IPFS verification with retries
+ *
+ * Use retries because for some reason IPFS (very rarely) fails due to some non-deterministic error. Seems to be with the verification step; the files are correct.
+ * Wait fixed timeout between each retry, hopefully addresses IPFS non-deterministic errors
+ */
+const _dirCIDIPFSVerificationWithRetries = async function (req, resizeResp, dirCID, retriesLeft = IMAGE_UPLOAD_IPFS_VERIFICATION_RETRY_COUNT) {
+  const ipfs = req.app.get('ipfsLatestAPI')
+
+  // build ipfs add array
+  let ipfsAddArray = []
+  try {
+    await Promise.all(resizeResp.files.map(async function (file) {
+      const fileBuffer = await fs.readFile(file.storagePath)
+      ipfsAddArray.push({
+        path: file.sourceFile,
+        content: fileBuffer
+      })
+    }))
+  } catch (e) {
+    throw new Error(`Failed to build ipfs add array for dirCID ${dirCID} ${e}`)
+  }
+
+  // Re-compute dirCID from all image files to ensure it matches dirCID returned above
+  let ipfsAddRespArr
+  try {
+    const ipfsAddResp = await ipfs.add(
+      ipfsAddArray,
+      {
+        pin: false,
+        onlyHash: true,
+        timeout: 1000
+      }
+    )
+    ipfsAddRespArr = []
+    for await (const resp of ipfsAddResp) {
+      ipfsAddRespArr.push(resp)
+    }
+  } catch (e) {
+    // If ipfs.add op fails, log error and move on, since this is an ipfs error and not an image upload error
+    req.logger.info(`Error calling ipfs.add on dir to re-compute dirCID ${dirCID} ${e}`)
+  }
+
+  // Ensure actual and expected dirCIDs match
+  const expectedDirCID = ipfsAddRespArr[ipfsAddRespArr.length - 1].cid.toString()
+  if (expectedDirCID !== dirCID) {
+    if (retriesLeft > 0) {
+      req.logger.error(`Image file validation failed - dirCIDs do not match for dirCID=${dirCID} expectedCID=${expectedDirCID}. ${retriesLeft} retries remaining out of ${IMAGE_UPLOAD_IPFS_VERIFICATION_RETRY_COUNT}. Retrying...`)
+      await timeout(IMAGE_UPLOAD_IPFS_VERIFICATION_RETRY_TIMEOUT_MS)
```

Comment:
is this timeout necessary? is ipfs.add dependent on some wait period?

Reply:
good point, doesn't seem necessary - removed